pagerduty/incident.go

```diff
@@ -0,0 +1,154 @@
+	LastStatusChangeBy   *IncidentAttributeReference `json:"last_status_change_by,omitempty"`
+	FirstTriggerLogEntry *IncidentAttributeReference `json:"first_trigger_log_entry,omitempty"`
+	EscalationPolicy     *EscalationPolicyReference  `json:"escalation_policy,omitempty"`
+	Teams                []*TeamReference            `json:"teams,omitempty"`
+	Urgency              string                      `json:"urgency,omitempty"`
+}
+
+type AlertCounts struct {
+	All       int `json:"all"`
+	Resolved  int `json:"resolved"`
+	Triggered int `json:"triggered"`
+}
+
+type PendingAction struct {
+	At   string `json:"at"`
+	Type string `json:"type"`
+}
+
+type IncidentAssignment struct {
+	At       string        `json:"at"`
+	Assignee UserReference `json:"assignee"`
+}
+
+type IncidentAcknowledgement struct {
+	At           string                     `json:"at"`
+	Acknowledger IncidentAttributeReference `json:"acknowledger"`
+}
+
+// IncidentPayload represents an incident.
+type IncidentPayload struct {
+	Incident *Incident `json:"incident,omitempty"`
+}
+
+// ManageIncidentsPayload represents a payload with a list of incidents data.
+type ManageIncidentsPayload struct {
+	Incidents []*Incident `json:"incidents,omitempty"`
+}
+
+// ListIncidentsOptions represents options when listing incidents.
+type ListIncidentsOptions struct {
+	Limit       int      `url:"limit,omitempty"`
+	Offset      int      `url:"offset,omitempty"`
+	Total       int      `url:"total,omitempty"`
+	DateRange   string   `url:"date_range,omitempty"`
+	IncidentKey string   `url:"incident_key,omitempty"`
+	Include     []string `url:"include,omitempty,brackets"`
+	ServiceIDs  []string `url:"service_ids,omitempty,brackets"`
+	Since       string   `url:"since,omitempty"`
+	SortBy      []string `url:"sort_by,omitempty,brackets"`
+	Statuses    []string `url:"statuses,omitempty,brackets"`
+	TeamIDs     []string `url:"team_ids,omitempty,brackets"`
+	TimeZone    string   `url:"time_zone,omitempty"`
+	Until       string   `url:"until,omitempty"`
+	Urgencies   []string `url:"urgencies,omitempty,brackets"`
+	UserIDs     []string `url:"user_ids,omitempty,brackets"`
+}
+
+// ManageIncidentsOptions represents options when listing incidents.
+type ManageIncidentsOptions struct {
+	Limit  int `url:"limit,omitempty"`
+	Offset int `url:"offset,omitempty"`
+	Total  int `url:"total,omitempty"`
+}
+
+// ListIncidentsResponse represents a list response of incidents.
+type ListIncidentsResponse struct {
+	Limit     int         `json:"limit,omitempty"`
+	More      bool        `json:"more,omitempty"`
+	Offset    int         `json:"offset,omitempty"`
+	Total     int         `json:"total,omitempty"`
+	Incidents []*Incident `json:"incidents,omitempty"`
+}
+
+type ManageIncidentsResponse ListIncidentsResponse
+
+// List lists existing incidents.
+func (s *IncidentService) List(o *ListIncidentsOptions) (*ListIncidentsResponse, *Response, error) {
```

Comment:
@imjaroiswebdev Will we need to use pagination to grab the full list of incidents?